src/components/Dashboard.tsx

```diff
@@ -94,38 +85,89 @@ const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template })
   )
 }
 
-const FilterComponent: React.FC<{
+const PanelComponent: React.FC<{
   template: TemplateDetails
-  filter: Filter
-  setFiltersCount: Function
-}> = ({ template, filter, setFiltersCount: setLoadingFilters }) => {
+  filters: Filter[]
+}> = ({ template, filters }) => {
   const templateType = template.code
-  const { loading, applicationCount } = useListApplications({
-    type: templateType,
-    perPage: 1,
-    ...filter.query,
-  })
+  const [loadedFiltersCount, setLoadedFiltersCount] = useState(0)
+  const [totalMatchFilter, setTotalMatchFilter] = useState<{ [key: number]: number }>(
+    filters.reduce((totalPerFilter, filter) => ({ ...totalPerFilter, [filter.id]: 0 }), {})
+  )
+
+  const arrayFilters = filters.reduce((arrayFilters: { [key: string]: string }[], element) => {
+    if (typeof element.query === 'object') {
+      const queryObj = element.query as { [key: string]: string }
+      arrayFilters.push(queryObj)
+    }
+    return arrayFilters
+  }, [])
+
+  const queryMultiFilters = {
+    templateCode: { equalToInsensitive: templateType },
+    ...constructOrObjectFilters(arrayFilters),
+  }
+
+  const { loading, applications } = useListApplications(
+    {}, // Passing empty to use already construct GraphQL query on queryMultiFilters
+    queryMultiFilters
+  )
 
   useEffect(() => {
-    if (!loading) setLoadingFilters((currentCount: number) => currentCount + 1)
-  }, [loading])
+    if (applications) {
+      filters.forEach(({ id, query }) => {
+        if (typeof query === 'object') {
+          const queryObj = query as { [key: string]: string }
+          const filteredApplications = Object.entries(applications).filter(([_, application]) => {
+            // Each filter is currently delimited to a single check!
+            const key = Object.keys(queryObj)[0]
+            const value = Object.values(queryObj)[0]
 
-  const applicationListUserRole =
+            switch (key) {
+              case 'outcome':
+                return application.outcome === value
```

Comment:
Is this the same as the more common:
```
 case 'outcome':
    application.outcome === value
    break
```

Personally I prefer this syntax (cos you're not really returning anything, and we generally expect `switch` blocks to have `breaks` when not `return`ing), but I guess if it works, it's probably fine (it's at least concise). Up to you.

If it's *not* the same, can you explain the difference?